src/library/browse/browsethread.cpp

```diff
@@ -154,15 +153,16 @@ void BrowseThread::populateModel() {
 
         const QString filepath = fileIt.next();
         {
-            TrackPointer pTrack =
-                GlobalTrackCache::instance().resolve(
-                        filepath, thisPath.token()).getTrack();
-            // The GlobalTrackCache is unlocked instantly even if a new track object
-            // has been created and inserted into the cache. Newly created track
-            // objects will only contain a reference of the corresponding file,
-            // but not any metadata, yet. This reduces lock contention on the
-            // global track cache.
-
+            // We need to create a temporary track object and cannot
+            // use the GlobalTrackCache, because cached tracks must
+            // live in the main thread. There is a tiny chance that
+            // the file is removed (after exporting metadata) just
+            // in the moment when reading metadata from it.
+            // TODO: Resolve track object through GlobalTrackCache
+            // after multi-threaded library management is in place
+            // to ensure that only a single track object exists for
+            // each file.
+            TrackPointer pTrack = Track::newTemporary(filepath, thisPath.token());
```

Comment:
Why need they live in the main thread? Why can't this temporary track not live in the main thread as well?

Reply:
This temporary will be destroyed at the end of the scope. If we fetch an existing track object from the cache we might be the last owner when leaving the current scope which is executed in the browse thread. This might trigger a database update which is only allowed in the main thread.

Ideally database updates and metadata exports should be performed asynchronously in a worker thread by enqueueing tracks prior to deletion in GlobalTrackCache. Currently the database updates and metadata exports are executed immediately in the thread context that invokes the destructor of TrackPointer.

While metadata is exported by a worker thread the GlobalTrackCache must still ensure that no one else is accessing the corresponding file as it already does.

Reply:
OK, how about this:

// We need to create a temporary track object because 
// we want to see the file meta-data and not from the database.
// Using the GlobalTrackCache here would override the database 
// metadata with the file metadata.